pkg/operation/botanist/component/vpa/updater.go

```diff
@@ -0,0 +1,213 @@
+	}
+
+	if v.values.ClusterType == ClusterTypeSeed {
+		serviceAccount := v.emptyServiceAccount(updater)
+		configs = append(configs,
+			resourceConfig{obj: serviceAccount, class: application, mutateFn: func() { v.reconcileUpdaterServiceAccount(serviceAccount) }},
+			resourceConfig{obj: deployment, class: runtime, mutateFn: func() { v.reconcileUpdaterDeployment(deployment, &serviceAccount.Name) }},
+		)
+	} else {
+		configs = append(configs,
+			resourceConfig{obj: deployment, class: runtime, mutateFn: func() { v.reconcileUpdaterDeployment(deployment, nil) }},
+		)
+	}
+
+	return configs
+}
+
+func (v *vpa) reconcileUpdaterServiceAccount(serviceAccount *corev1.ServiceAccount) {
+	serviceAccount.Labels = getRoleLabel()
+	serviceAccount.AutomountServiceAccountToken = pointer.Bool(false)
+}
+
+func (v *vpa) reconcileUpdaterClusterRole(clusterRole *rbacv1.ClusterRole) {
+	clusterRole.Labels = getRoleLabel()
+	clusterRole.Rules = []rbacv1.PolicyRule{
+		{
+			APIGroups: []string{"apps", "extensions"},
+			Resources: []string{"replicasets"},
+			Verbs:     []string{"get"},
+		},
+		{
+			APIGroups: []string{""},
+			Resources: []string{"pods/eviction"},
+			Verbs:     []string{"create"},
+		},
+	}
+}
+
+func (v *vpa) reconcileUpdaterClusterRoleBinding(clusterRoleBinding *rbacv1.ClusterRoleBinding, clusterRole *rbacv1.ClusterRole, serviceAccountName string) {
+	clusterRoleBinding.Labels = getRoleLabel()
+	clusterRoleBinding.Annotations = map[string]string{resourcesv1alpha1.DeleteOnInvalidUpdate: "true"}
+	clusterRoleBinding.RoleRef = rbacv1.RoleRef{
+		APIGroup: rbacv1.GroupName,
+		Kind:     "ClusterRole",
+		Name:     clusterRole.Name,
+	}
+	clusterRoleBinding.Subjects = []rbacv1.Subject{{
+		Kind:      rbacv1.ServiceAccountKind,
+		Name:      serviceAccountName,
+		Namespace: v.serviceAccountNamespace(),
+	}}
+}
+
+func (v *vpa) reconcileUpdaterDeployment(deployment *appsv1.Deployment, serviceAccountName *string) {
+	deployment.Labels = v.getDeploymentLabels(updater)
+	deployment.Spec = appsv1.DeploymentSpec{
+		Replicas:             &v.values.Updater.Replicas,
+		RevisionHistoryLimit: pointer.Int32(2),
+		Selector:             &metav1.LabelSelector{MatchLabels: getAppLabel(updater)},
+		Template: corev1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: utils.MergeStringMaps(getAllLabels(updater), map[string]string{
+					v1beta1constants.LabelNetworkPolicyFromPrometheus: v1beta1constants.LabelNetworkPolicyAllowed,
+				}),
+			},
+			Spec: corev1.PodSpec{
+				Containers: []corev1.Container{{
+					Name:            "updater",
+					Image:           v.values.Updater.Image,
+					ImagePullPolicy: corev1.PullIfNotPresent,
+					Command:         []string{"./updater"},
+					Args: []string{
+						"--min-replicas=1",
+						fmt.Sprintf("--eviction-tolerance=%f", pointer.Float64Deref(v.values.Updater.EvictionTolerance, gardencorev1beta1.DefaultEvictionTolerance)),
```

Comment:
Nit: This value (and most other float values) was previously formatted as `1` (no trailing zeroes), but is now formatted as `1.00000`, although it certainly doesn't matter much, it's a bit nicer to avoid such zeroes, which should be possible with different formatting parameters.

Reply:
Which formatting parameters do you suggest? Does this matter at all? In the end, it's now "more precise" than before